doc/developer/feature-branch.rst

```diff
@@ -0,0 +1,109 @@
+Specific Feature Development Branch
+===================================
+
+This document describes the process to request and manage a dedicated branch
+on the FRR GitHub repo for the development of new features.
+
+Why dedicated branch to develop new feature?
+--------------------------------------------
+
+Some new features need to be developed in a collaborative way
+e.g. IS-IS Segment Routing, BGP-LS, PCC … and this specific development will
+take time.
+
+Providing a common Git reference will ease cooperation between partners.
+Indeed, it is hard to synchronise independent Git Repo. Only main FRR GitHub
+repo are known by everybody and private forks to create Pull Request are not
+known by everybody. In addition, there no space on FRR web site or wiki to
+publicly announce that someone is working on a given feature. FRR community
+is, in general, aware about the development of a feature when Pull Request is
+published, which is be too late for good cooperation.
+
+Feature Branch characteristics
+------------------------------
+
+A dedicated branch for a specific feature collaborative development is created
+as follow in the GitHub FRR repo (http://github.com/FRRouting/frr):
+
+- dev-XXX where XXX is the acronyme of the feature e.g. dev-bgpls
+- Feature branch is not linked to CI/CD chain, so will not trigger any tests
+  not compilation when a Pull Request is submitted to this branch
+- The README.md file from master branch is modified as follow:
```

Comment:
I don't think this belongs in the readme, if anything it belongs on a new wiki page.

Reply:
I proposed this modification following a remark from Martin about a possible confusion for peoples unaware about our internal process. But, this could take another form e.g. a dedicated Markdown file to explain this is a development branch. The main idea is to be sure that peoples will not raise an issue about a development branch.